lib/routes/account.js

```diff
@@ -1021,50 +1045,78 @@ module.exports = function (
         validate: {
           payload: {
             uid: isA.string().max(32).regex(HEX_STRING).required(),
-            code: isA.string().min(32).max(32).regex(HEX_STRING).required()
+            code: isA.string().min(32).max(64).regex(HEX_STRING).required()
           }
         }
       },
       handler: function (request, reply) {
-        log.begin('Account.RecoveryEmailVerify', request)
         var uid = request.payload.uid
         var code = Buffer(request.payload.code, 'hex')
-        db.account(Buffer(uid, 'hex'))
-          .then(
-            function (account) {
-              // If the account is already verified, they may be e.g.
-              // clicking a stale link.  Silently succeed.
-              if (account.emailVerified) {
-                log.increment('account.already_verified')
-                return true
-              }
-              if (!butil.buffersAreEqual(code, account.emailCode)) {
-                throw error.invalidVerificationCode()
-              }
-              log.timing('account.verified', Date.now() - account.createdAt)
-              log.event('verified', { email: account.email, uid: account.uid, locale: account.locale })
-              log.increment('account.verified')
 
-              // send a push notification to all devices that the account changed
-              push.notifyUpdate(uid)
+        if(code.length === 32) {
```

Comment:
@shane-tomlinson Looks like I forgot to update this. It requires a token with uid set, to validate that tokenVerificationId belongs to the uid, [ref](https://github.com/mozilla/fxa-auth-db-mysql/blob/phil/fxa-83/fxa-auth-db-server/test/backend/db_tests.js#L349). Will be updating this section soon.